File: test_feature/planning/impl/src/main/java/com/github/klee0kai/stone/test_feature/planning/PlanningRun.java
package com.github.klee0kai.stone.test_feature.planning;

import com.github.klee0kai.stone.test_feature.finance.model.WorkCalendar;
import com.github.klee0kai.stone.test_feature.hr.department.SecurityDepartment;
import com.github.klee0kai.stone.test_feature.planning.di.PlanningComponent;
import com.github.klee0kai.stone.test_feature.planning.project.LogisticProject;

import javax.inject.Inject;

public class PlanningRun {

    @Inject
    public WorkCalendar workCalendar;

    @Inject
    public LogisticProject logisticProject;

    @Inject
    public SecurityDepartment securityDepartment;

    private final PlanningComponent component;

    public PlanningRun(PlanningComponent component) {
        this.component = component;
    }

    public void start() {
        component.inject(this);
    }

}

File: test_feature/planning/impl/src/main/java/com/github/klee0kai/stone/test_feature/planning/PlanningRun.kt
package com.github.klee0kai.stone.test_feature.planning

import com.github.klee0kai.stone.test_feature.finance.model.WorkCalendar
import com.github.klee0kai.stone.test_feature.hr.department.SecurityDepartment
import com.github.klee0kai.stone.test_feature.planning.di.PlanningComponent
import com.github.klee0kai.stone.test_feature.planning.project.LogisticProject
import javax.inject.Inject

class PlanningRun(private val component: PlanningComponent) {

    @Inject
    var workCalendar: WorkCalendar? = null

    @Inject
    var logisticProject: LogisticProject? = null

    @Inject
    var securityDepartment: SecurityDepartment? = null

    fun start() {
        component.inject(this)
    }
}
